o1vm/src/cli/mod.rs

```diff
@@ -0,0 +1,14 @@
+use clap::Parser;
+
+pub mod cannon;
+
+#[derive(Parser, Debug, Clone)]
+#[command(
+    name = "o1vm",
+    version = "0.1",
+    about = "o1vm - a generic purpose zero-knowledge virtual machine"
+)]
+pub enum Commands {
+    #[command(subcommand)]
+    Cannon(cannon::Cannon),
```

Comment:
I guess "non-cannon" version will appear here. Waiting to see the other subcommands :).